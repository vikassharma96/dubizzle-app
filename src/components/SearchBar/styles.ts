import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';
import defaultStyles from '../../styles/defaultStyles';
import {fontSizes} from '../../styles/fonts';

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    marginStart: '3%',
    paddingVertical: '2%',
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'row',
    height: '100%',
  },
  searchBar: {
    width: '80%',
    flexDirection: 'row',
    backgroundColor: colors.background,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingHorizontal: '3%',
    height: '100%',
  },
  input: {
    marginStart: '2%',
    alignItems: 'center',
    paddingTop: 0,
    paddingBottom: 0,
    width: '90%',
    ...defaultStyles.text,
    fontSize: fontSizes.fontSize_12,
  },
});

export default styles;
