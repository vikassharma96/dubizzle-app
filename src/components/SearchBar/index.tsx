import React, {useState} from 'react';
import {TouchableOpacity} from 'react-native';
import {Keyboard, TextInput, View} from 'react-native';
import Icon from 'react-native-vector-icons/Octicons';
import styles from './styles';

type IProps = {
  onSearchUserGist: (userName: string) => void;
  onResetUserGist: () => void;
};

const SearchBar = ({onSearchUserGist, onResetUserGist}: IProps) => {
  const [isFocused, setIsFocused] = useState<boolean>(false);
  const [userName, setUserName] = useState<string>('');

  const onSearch = () => {
    Keyboard.dismiss();
    onSearchUserGist(userName.trim());
  };

  const onReset = () => {
    Keyboard.dismiss();
    setUserName('');
    setIsFocused(false);
    if (userName) {
      onResetUserGist();
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.searchBar}>
        <Icon name="search" color="black" size={15} />
        <TextInput
          style={styles.input}
          numberOfLines={1}
          placeholder="Search Gists for the username"
          value={userName}
          onChangeText={setUserName}
          returnKeyType={'search'}
          onSubmitEditing={onSearch}
          onFocus={() => setIsFocused(true)}
        />
        {isFocused && (
          <TouchableOpacity onPress={onReset}>
            <Icon name="x" color="black" size={15} />
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

export default SearchBar;
