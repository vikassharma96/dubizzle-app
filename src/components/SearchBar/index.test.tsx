import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '.';

describe('@components/SearchBar', () => {
  const mockOnSearchUserGist = jest.fn();
  const mockOnResetUserGist = jest.fn();
  const component = (
    <SearchBar
      onSearchUserGist={mockOnSearchUserGist}
      onResetUserGist={mockOnResetUserGist}
    />
  );

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
