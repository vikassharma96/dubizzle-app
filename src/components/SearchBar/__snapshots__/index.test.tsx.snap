// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/SearchBar should match snapshots 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "flexGrow": 1,
      "height": "100%",
      "justifyContent": "flex-start",
      "marginStart": "3%",
      "paddingVertical": "2%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "space-evenly",
        "paddingHorizontal": "3%",
        "width": "80%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 15,
          },
          undefined,
          {
            "fontFamily": "Octicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <TextInput
      numberOfLines={1}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search Gists for the username"
      returnKeyType="search"
      style={
        {
          "alignItems": "center",
          "fontFamily": "Poppins-Medium",
          "fontSize": 12,
          "marginStart": "2%",
          "paddingBottom": 0,
          "paddingTop": 0,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
</View>
`;
